pkg/autoscaler/autoscaler.go

```diff
@@ -30,6 +30,11 @@ const (
 	// ActivatorPodName defines the pod name of the activator
 	// as defined in the metrics it sends.
 	ActivatorPodName string = "activator"
+
+	// minOutOfServiceTime is the threshold to consider a pod becomes out of
+	// service(lameducked). There is always a gap between current time and the
+	// point receiving the lastest stat from a pod.
+	minOutOfServiceTime time.Duration = time.Second
```

Comment:
Wondering if this should be a bit bigger. Maybe equal to the interval the autoscaler runs in or something, since we kinda expect metrics to arrive in that timeframe anyway.

Reply:
Is `outOfServiceThreshold` a more speaking name? Debating with myself 😆

Reply:
Renamed to `activeThreshold` since when we do sampling, we can only tell a pod is active if we just receive its metrics recently.

The `queue-proxy` is pushing metrics every 1 second and the autoscaler calculates the desired pods every 2 seconds. I think this should equal to metrics reporting interval, not the autoscaler calculation interval? Anyway we can tune this later.